metrics/resource_view_test.go

```diff
@@ -566,25 +585,38 @@ func TestStackDriverExports(t *testing.T) {
 				notReadyPodCountM.M(int64(3)))
 
 			records := []metricExtract{}
-			for record := range sdFake.published {
-				for _, ts := range record.TimeSeries {
-					records = append(records, metricExtract{
-						Name:   ts.Metric.Type,
-						Labels: ts.Resource.Labels,
-						Value:  ts.Points[0].Value.GetInt64Value(),
-					})
-					if strings.HasPrefix(ts.Metric.Type, "knative.dev/") {
-						if diff := cmp.Diff(ts.Resource.Type, metricskey.ResourceTypeKnativeRevision); diff != "" {
-							t.Errorf("Incorrect resource type for %q: (-want +got): %s", ts.Metric.Type, diff)
+		loop:
+			for {
+				select {
+				case record := <-sdFake.published:
+					for _, ts := range record.TimeSeries {
+						extracted := metricExtract{
+							Name:   ts.Metric.Type,
+							Labels: ts.Resource.Labels,
+							Value:  ts.Points[0].Value.GetInt64Value(),
+						}
+						// Override 'cluster-name' label to reset to a fixed value
+						if extracted.Labels["cluster_name"] != "" {
```

Comment:
Should it not be set if empty?

Reply:
This is using Golang defaults, so `mapVal["doesNotExist"]` will return the default value from the map `map[string]string`, which is `""`.

Reply:
Strictly theoretically there may be the value in the map with `""` value. So if this happens due to a bug it won't be caught. But 🤷 .